### PR TITLE
Added question/findByAuthor endpoint

### PR DESCRIPTION
**Pull request by maybee16**
https://trello.com/c/QIDGtxpO/95-surveyservice-add-question-findbyauthor-endpoint

### CODE REVIEW

File: src/SurveyService.Data/QuestionRepository.cs [hunk 1]
```diff
@@ -51,4 +59,39 @@ public Task<bool> DoesExistAsync(Guid questionId)
   {
     return _provider.Questions.AnyAsync(x => x.Id == questionId);
   }
+
+  public async Task<(List<DbQuestion>, int totalCount)> FindByAuthorAsync(FindByAuthorFilter filter)
+  {
+    IQueryable<DbQuestion> query = _provider.Questions.AsQueryable();
+
+    query = query.Where(q => q.CreatedBy == _httpContextAccessor.HttpContext.GetUserId()
```
Comment (vorobevamv) [inline src/SurveyService.Data/QuestionRepository.cs]: Move getting author Id to command (_httpContextAccessor.HttpContext.GetUserId());
and add to method parameters Id: ...FindByAuthorAsync(FindByAuthorFilter filter, Guid authorId)

File: src/SurveyService.Data/QuestionRepository.cs [hunk 2]
```diff
@@ -51,4 +59,39 @@ public Task<bool> DoesExistAsync(Guid questionId)
...
+
+    if (filter.IsAscendingSort.HasValue)
+    {
+      query = filter.IsAscendingSort.Value
+        ? query.OrderBy(q => q.Content)
+        : query.OrderByDescending(q => q.Content);
+    }
+    else
+    {
+      query = query.OrderByDescending(q => q.CreatedAtUtc);
```
Comment (vorobevamv) [inline src/SurveyService.Data/QuestionRepository.cs]: add OrderByDescending(q => q.CreatedAtUtc) to line 65 and remove 'else' (76-79)

File: src/SurveyService.Data/QuestionRepository.cs [hunk 3]
```diff
@@ -51,4 +59,39 @@ public Task<bool> DoesExistAsync(Guid questionId)
...
+    if (filter.IsActive.HasValue)
+    {
+      query = filter.IsActive.Value
+        ? query.Where(q => q.IsActive)
+        : query.Where(q => !q.IsActive);
+    }
+
+    if (filter.IncludeGroup)
+    {
+      query = query.Include(q => q.Group);
```
Comment (vorobevamv) [inline src/SurveyService.Data/QuestionRepository.cs]: line 65: ...&& q.GroupId == null...
what for this IF?
IncludeGroup = false -> return only single questions;
IncludeGroup = true -> return single questions and groups;

File: src/SurveyService.Data/GroupRepository.cs [hunk 4]
```diff
@@ -27,4 +37,33 @@ public GroupRepository(IDataProvider provider)
     
     return dbGroup.Id;
   }
+
+  public async Task<(List<DbGroup>, int totalCount)> FindByAuthorAsync(FindByAuthorFilter filter)
+  {
+    IQueryable<DbGroup> query = _provider.Groups.AsQueryable();
+
+    query = query.Where(g => g.CreatedBy == _httpContextAccessor.HttpContext.GetUserId());
```
Comment (vorobevamv) [inline src/SurveyService.Data/GroupRepository.cs]: the same:
_httpContextAccessor.HttpContext.GetUserId() use in command; add to method parameters Guid AuthorId;
add to line 45 OrderByDescending(g => g.CreatedAtUtc) and remove ELSE (53-56)

File: src/SurveyService.Data/QuestionRepository.cs (hunk 1 above)
Comment (maybee16) [reply src/SurveyService.Data/QuestionRepository.cs]: Done

File: src/SurveyService.Data/QuestionRepository.cs (hunk 2 above)
Comment (maybee16) [reply src/SurveyService.Data/QuestionRepository.cs]: Done

File: src/SurveyService.Data/QuestionRepository.cs (hunk 3 above)
Comment (maybee16) [reply src/SurveyService.Data/QuestionRepository.cs]: Done

File: src/SurveyService.Data/GroupRepository.cs (hunk 4 above)
Comment (maybee16) [reply src/SurveyService.Data/GroupRepository.cs]: Done

**Review by vorobevamv — APPROVED** [review]

File: src/SurveyService/Controllers/QuestionController.cs [hunk 5]
```diff
@@ -18,4 +19,12 @@ public class QuestionController : ControllerBase
   {
     return await command.ExecuteAsync(request);
   }
+
+  [HttpGet("find")]
+  public async Task<FindResultResponse<FindByAuthorResultInfo>> FindAsync(
+    [FromServices] IFindByAuthorCommand command,
```
Comment (ndoriomedov) [inline src/SurveyService/Controllers/QuestionController.cs]: rename to IFindQuestionsCommand

File: src/SurveyService/Controllers/QuestionController.cs [hunk 6]
```diff
@@ -18,4 +19,12 @@ public class QuestionController : ControllerBase
...
+    [FromQuery] FindByAuthorFilter filter)
```
Comment (ndoriomedov) [inline src/SurveyService/Controllers/QuestionController.cs]: rename to FindQuestionsFilter

**Review by ndoriomedov — APPROVED** [review]

File: src/SurveyService/Controllers/QuestionController.cs (hunk 5 above)
Comment (maybee16) [reply src/SurveyService/Controllers/QuestionController.cs]: Done

File: src/SurveyService/Controllers/QuestionController.cs (hunk 6 above)
Comment (maybee16) [reply src/SurveyService/Controllers/QuestionController.cs]: Done

**Review by VLiamin — APPROVED** [review]

**Review by AlfaRedDog — APPROVED** [review]

**Pull request merged**